CSharp/Library/Microsoft.Bot.Connector.Shared/AttachmentEx.cs

```diff
@@ -0,0 +1,34 @@
+﻿namespace Microsoft.Bot.Connector
+{
+    using System;
+
+    public partial class Attachment : IEquatable<Attachment>
+    {
+        public bool Equals(Attachment other)
+        {
+            return other != null
+                && this.ContentType == other.ContentType
+                && this.ContentUrl == other.ContentUrl
+                && object.Equals(this.Content, other.Content)
+                && object.Equals(this.Name, other.Name)
+                && object.Equals(this.ThumbnailUrl, other.ThumbnailUrl);
+
+        }
+
+        public override bool Equals(object other)
+        {
+            return this.Equals(other as Attachment);
+        }
+
+        public override int GetHashCode()
+        {
+            var code = this.ContentType.GetHashCode()
+                ^ this.ContentUrl.GetHashCode()
+                ^ (this.Content == null ? 13 : this.Content.GetHashCode())
```

Comment:
Another example is like this:

```
int hash = 13;
hash = (hash * 7) + field1.GetHashCode();
hash = (hash * 7) + field2.GetHashCode();
...
return hash;
```

from here: http://stackoverflow.com/questions/371328/why-is-it-important-to-override-gethashcode-when-equals-method-is-overridden?rq=1

But, I suppose they could be any arbitrary number since we are doing an xor.